crates/cheatcodes/src/inspector.rs

```diff
@@ -308,6 +505,227 @@ impl Cheatcodes {
+                    ));
+                }
+
+                ecx.env.tx.caller = broadcast.new_origin;
+
+                if ecx.journaled_state.depth() == broadcast.depth {
+                    input.set_caller(broadcast.new_origin);
+                    let is_fixed_gas_limit = check_if_fixed_gas_limit(ecx, input.gas_limit());
+
+                    let account = &ecx.journaled_state.state()[&broadcast.new_origin];
+                    self.broadcastable_transactions.push_back(BroadcastableTransaction {
+                        rpc: ecx.db.active_fork_url(),
+                        transaction: TransactionRequest {
+                            from: Some(broadcast.new_origin),
+                            to: None,
+                            value: Some(input.value()),
+                            input: TransactionInput::new(input.init_code().clone()),
+                            nonce: Some(account.info.nonce),
+                            gas: if is_fixed_gas_limit {
+                                Some(input.gas_limit() as u128)
+                            } else {
+                                None
+                            },
+                            ..Default::default()
+                        },
+                    });
+
+                    input.log_debug(self, &input.scheme().unwrap_or(CreateScheme::Create));
+                }
+            }
+        }
+
+        // Allow cheatcodes from the address of the new contract
+        let address = input.allow_cheatcodes(self, ecx);
+
+        // If `recordAccountAccesses` has been called, record the create
+        if let Some(recorded_account_diffs_stack) = &mut self.recorded_account_diffs_stack {
+            recorded_account_diffs_stack.push(vec![AccountAccess {
+                chainInfo: crate::Vm::ChainInfo {
+                    forkId: ecx.db.active_fork_id().unwrap_or_default(),
+                    chainId: U256::from(ecx.env.cfg.chain_id),
+                },
+                accessor: input.caller(),
+                account: address,
+                kind: crate::Vm::AccountAccessKind::Create,
+                initialized: true,
+                oldBalance: U256::ZERO, // updated on (eof)create_end
+                newBalance: U256::ZERO, // updated on (eof)create_end
+                value: input.value(),
+                data: input.init_code().clone(),
+                reverted: false,
+                deployedCode: Bytes::new(), // updated on (eof)create_end
+                storageAccesses: vec![],    // updated on (eof)create_end
+                depth: ecx.journaled_state.depth(),
+            }]);
+        }
+
+        None
+    }
+
+    // common create_end functionality for both legacy and EOF.
+    fn create_end_common<DB, Input>(
+        &mut self,
+        ecx: &mut EvmContext<DB>,
+        input: Input,
+    ) -> CommonEndOutcome
+    where
+        DB: DatabaseExt,
+        Input: CommonEndInput,
```

Comment:
to me this makes sense, 
perhaps @DaniPopes perhaps has concerns re codegen, but should be okay I think

Reply:
I think it's better to have a common enum rather than a trait. Honestly this kinda sucks because we want to treat both as the same. I would rather unify these types as much as possible and only have one inspector function. @rakita 

For example eofcreate input can be another variant in `CreateScheme`, and its outcome is the same as `CreateOutcome` but a different type.

Reply:
Future idea is to have generic Frames, that would mean both Crate and Create2 have their own implhttps://github.com/bluealloy/revm/issues/996#issuecomment-1975650177 .

It is not great to merge them all in one and that is why it is separate type as when custom Frames come it will be hard to integrate. That is why i decided to add additional functions, even thought they are similar.